### PR TITLE
Implementing `IndexToolsAdapter` for ES7.

### PR DESCRIPTION
**Pull request by dennisoelkers**
## Motivation and Context

This PR is implementing the `IndexToolsAdapter` for ES7.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/SearchRequestFactory.java [hunk 1]
```diff
@@ -73,27 +88,51 @@ public SearchSourceBuilder create(SearchesConfig config) {
...
-            bfb = QueryBuilders.boolQuery()
-                    .must(TimeRangeQueryFactory.create(range));
-        }
+    private void applyStreamsFilter(BoolQueryBuilder filteredQueryBuilder, SearchCommand command) {
+        command.streams()
+                .map(this::buildStreamIdFilter)
+                .ifPresent(filteredQueryBuilder::filter);
+    }
+
+    private BoolQueryBuilder buildStreamIdFilter(Set<String> streams) {
```
Comment (alex-konn) [inline graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/SearchRequestFactory.java:105]: I think we should centralize this logic somewhere. Equivalent methods are also contained in `IndexToolsAdapterES6`, `indexToolsAdapterES7`, and `SearchesAdapterES6`.

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/IndexToolsAdapterES7.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+    @Override
+    public Map<DateTime, Map<String, Long>> fieldHistogram(String fieldName, Set<String> indices, Optional<Set<String>> includedStreams, long interval) {
+        final BoolQueryBuilder queryBuilder = buildStreamIdFilter(includedStreams);
+
+        final FilterAggregationBuilder the_filter = AggregationBuilders.filter(AGG_FILTER, queryBuilder)
+                .subAggregation(AggregationBuilders.dateHistogram(AGG_DATE_HISTOGRAM)
+                        .field("timestamp")
+                        .subAggregation(AggregationBuilders.terms(AGG_MESSAGE_FIELD).field(fieldName))
+                        .interval(interval)
```
[the remark this answers is not in the document]
Comment (dennisoelkers) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/IndexToolsAdapterES7.java]: :heavy_check_mark:

File: graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/SearchRequestFactory.java (hunk 1 above)
Comment (dennisoelkers) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/SearchRequestFactory.java:105]: Same here. I would prefer to do this in a separate PR though, as it is non-trivial to make it usable across multiple storage driver modules.

Comment (alex-konn) [reply graylog-storage-elasticsearch7/src/main/java/org/graylog/storage/elasticsearch7/SearchRequestFactory.java:105]: :+1:

**Pull request merged**